PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -134,6 +182,34 @@ class PayPalWebCheckoutClient internal constructor(
         listener?.onPayPalWebCanceled()
     }
 
+    private fun deliverVaultSuccess() {
+        val deepLinkUrlResult = browserSwitchResult?.deepLinkUrl
+        val requestMetadata = browserSwitchResult?.requestMetadata
+
+        if (deepLinkUrlResult != null && requestMetadata != null) {
+            val isFailure = deepLinkUrlResult.path?.contains(VAULT_CANCEL_URL_PATH) ?: false
+            if (isFailure) {
+                vaultListener?.onPayPalWebVaultFailure(PayPalWebCheckoutError.malformedResultError)
+            } else {
+                val approvalTokenId =
+                    deepLinkUrlResult.getQueryParameter(DEEP_LINK_PARAM_APPROVAL_TOKEN_ID)
+                val approvalSessionId =
+                    deepLinkUrlResult.getQueryParameter(DEEP_LINK_PARAM_APPROVAL_SESSION_ID)
+                val result = PayPalWebCheckoutVaultResult(approvalTokenId, approvalSessionId)
```

Comment:
Oh we need the approvalSessionId? What is that for?

Reply:
I'm not 100% sure. I honestly just parse these values out because they're there.

Reply:
Without that value, the results from the vault are same for card and PayPal unless
it is decided that we return more values for card vaulting.

Reply:
Yeah for this PR we have two vault result types, `PayPalWebCheckoutVaultResult` and `CardVaultResult`. Are y'all using a single result type on iOS?

Reply:
No I kept it different because card endpoint sends a lot of info and I thought there might be request to include more data in the future. But as of now, what we are returning is same except for the approval session id.

Reply:
I ended up simplifying to `approvalSessionId` only on Android.